client/pages/quiz/[slug].js

```diff
@@ -0,0 +1,166 @@
+/*
+  This page will load at the url "/quiz/:slug"
+*/
+
+import React, { useState, useEffect } from "react";
+// import useSWR from 'swr';// <-- uncomment when DB is ready
+import { useRouter } from "next/router";
+// import apiRoutes from '../../api-routes';// <-- uncomment when DB is ready
+import { dummyData } from "../dummy-quiz-data";
+// import DisplayMessage from '../../components/shared/DisplayMessage';
+import {
+  QuizHeader,
+  AnswersTileSection,
+  NextQuestionBtnContainer,
+  AnswerTileContainerLink,
+} from "../../components/quizSingle/styles";
+import { PageHeader } from "../../components/shared/styles";
+import Header from "../../components/header/Header";
+import QuestionHeader from "../../components/quizSingle/QuestionHeader";
+import AnswerTileContainer from "../../components/quizSingle/AnswerTileContainer";
+import NextQuestionBtn from "../../components/quizSingle/NextQuestionBtn";
+import { ContentWrapper } from "../../components/footer/styles";
+
+export default function Quiz() {
+  const router = useRouter();
+
+  // const fetcher = url => fetch(url).then(res => res.json());
+  // const {data, error} = useSWR(apiRoutes.getAllQuizzes, fetcher);
+  const [allSubjectQuizzes, setAllSubjectQuizzes] = useState([]);
+  const [chosenQuiz, setChosenQuiz] = useState({});
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [allQuestionsCount, setAllQuestionsCount] = useState(0);
+  const [currentQuestion, setCurrentQuestion] = useState({});
+  const [currentQuestionAnswers, setCurrentQuestionAnswers] = useState([]);
+  const [selectedAnswers, setSelectedAnswers] = useState({});
+
+  // When id is selected and quizzes are loaded,
+  // find quiz by id and save it to state
+  useEffect(() => {
+    if (router.query.slug) {
+      const filtered = allSubjectQuizzes.filter(quiz => {
+        return String(quiz.id) === String(router.query.slug);
+      });
+      const filteredQuiz = filtered.length ? filtered[0] : {};
+      setChosenQuiz(filteredQuiz);
+    }
+  }, [router.query.slug, allSubjectQuizzes]);
+
+  // ** Uncomment when DB is ready
+  // Set data returned from SWR in state
+  // useEffect(() => {
+  //   if (data) {
+  //     setAllSubjectQuizzes(data);
+  //   }
+  // }, [data]);
+
+  // Set Dummy Data into state until DB is ready
+  useEffect(() => {
+    if (dummyData) {
+      setAllSubjectQuizzes(dummyData);
+    }
+  }, [dummyData]);
+
+  // Set current quiz question to state
+  useEffect(() => {
+    if (chosenQuiz && chosenQuiz.questions && chosenQuiz.questions.length) {
+      setCurrentQuestion(chosenQuiz.questions[currentQuestionIndex]);
+      setAllQuestionsCount(chosenQuiz.questions.length);
+    }
+  }, [chosenQuiz, currentQuestionIndex]);
+
+  // Set current question answers to state
+  useEffect(() => {
+    if (currentQuestion) {
+      setCurrentQuestionAnswers(currentQuestion.answers);
+    }
+  }, [currentQuestion]);
+
+  const toggleSelectedAnswer = i => {
```

Comment:
This seems long and a bit complex. It could probably be a little simpler.

I would suggest something like this (could of course be changed):
`  const toggleSelectedAnswer = answer => {
    // Clone state object
    const selectedAnswersTmp = { ...selectedAnswers };

    // Get selected answer array
    const selectedAnswersArray = selectedAnswersTmp[currentQuestionIndex];

    // Set / remove answer
    if (!selectedAnswersArray) {
      selectedAnswersTmp[currentQuestionIndex] = [answer.id];
    } else if (selectedAnswersArray.find(id => id === answer.id)) {
      selectedAnswersTmp[currentQuestionIndex].splice(
        selectedAnswersArray.indexOf(answer.id),
        1
      );
    } else {
      selectedAnswersArray.push(answer.id);
    }

    // Save state
    setSelectedAnswers(selectedAnswersTmp);
  };`

Reply:
Also in the logic you want to handle when someone checks a different answer (changes their mind). Currently if I select an answer then choose another one they both look selected. 

There a different ways to accomplish this, reach out if you need some ideas! 

Another 'nice to have' would be if there isn't an answer chosen have the 'Next' button greyed out and not let them continue (we might change it to let people skip and come back later, but I think we want to keep it linear for now).

Reply:
@ZumDeWald The logic of this was intended as per issue #28 Multiple answer choices, so user can select more answers for question.
Though i agree it can be simplified a bit, ill try to use your suggestion @Soreg

Reply:
I used your suggested code in new commit